jobs/resource_job_test.go

```diff
@@ -429,6 +430,161 @@ func TestResourceJobCreateNWorkers(t *testing.T) {
 	assert.Equal(t, "789", d.Id())
 }
 
+func TestResourceJobCreateFromGitSource(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "POST",
+				Resource: "/api/2.1/jobs/create",
+				ExpectedRequest: JobSettings{
+					ExistingClusterID: "abc",
+					Tasks: []JobTaskSettings{
+						{
+							TaskKey: "b",
+							NotebookTask: &NotebookTask{
+								NotebookPath: "/GitSourcedNotebook",
+							},
+						},
+					},
+					Name:              "GitSourceJob",
+					MaxConcurrentRuns: 1,
+					GitSource: &GitSource{
+						Url:      "https://github.com/databrickslabs/terraform-provider-databricks",
+						Tag:      "0.4.8",
+						Provider: "gitHub",
+					},
+				},
+				Response: Job{
+					JobID: 789,
+				},
+			},
+			{
+				Method:   "GET",
+				Resource: "/api/2.1/jobs/get?job_id=789",
+				Response: Job{
+					JobID: 789,
+					Settings: &JobSettings{
+						// good enough for mock
+						ExistingClusterID: "abc",
+						Tasks: []JobTaskSettings{
+							{
+								TaskKey: "b",
+							},
+						},
+						Name:              "GitSourceJob",
+						MaxConcurrentRuns: 1,
+					},
+				},
+			},
+		},
+		Create:   true,
+		Resource: ResourceJob(),
+		HCL: `existing_cluster_id = "abc"
+		max_concurrent_runs = 1
+		name = "GitSourceJob"
+
+		git_source {
+			url = "https://github.com/databrickslabs/terraform-provider-databricks"
+			tag = "0.4.8"
+		}
+
+		task {
+			task_key = "b"
+
+			notebook_task {
+				notebook_path = "/GitSourcedNotebook"
+			}
+		}`,
+	}.Apply(t)
+	assert.NoError(t, err, err)
+	assert.Equal(t, "789", d.Id())
+}
+
+func resourceJobCreateFromGitSourceConflict(t *testing.T, conflictingArgs []string, gitSource string) {
+	var hclTemplate = `existing_cluster_id = "abc"
+		max_concurrent_runs = 1
+		name = "GitSourceJob"
+		
+		%s
+		
+		task {
+			task_key = "b"
+
+			notebook_task {
+				notebook_path = "/GitSourcedNotebook"
+			}
+		}
+	`
+	var hcl = fmt.Sprintf(hclTemplate, gitSource)
+	_, err := qa.ResourceFixture{
+		Create:   true,
+		Resource: ResourceJob(),
+		HCL:      hcl,
+	}.Apply(t)
+	assert.Error(t, err, err)
+
+	var expectedErr = "invalid config supplied."
```

Comment:
looks a bit strange with this concatenation. and a bit fragile to changing error message. generally i tend to create `var found bool` and have assertions in the loop, then `assert.True(t, found)` after the loop. it'll be more readable than concatenation.